### PR TITLE
Adding recipe based on ingredients

### PR DESCRIPTION
**Pull request by lena-A-Al**

### CODE REVIEW

File: src/components/RecipesFromIngredients/RecipesFromIngredients.tsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+  const [recievedRecipesInfo, setRecievedRecipesInfo] = useState({} as any);
+  const [loading, setloading] = useState(false);
+  /** selectors */
+  const { ingredients } = useSelector((state: RootState) => state.ingredients);
+  const { user } = useSelector((state: RootState) => state.user);
+
+  /**
+   * This function will combine all the ingredients names in a single string and pass it to the API endpoint
+   */
+  let ingredientsNames: any = [];
```
Comment (Bderiel) [inline src/components/RecipesFromIngredients/RecipesFromIngredients.tsx:28]: you want this array to be in useState if the component rerenders the array will clear. Or you could remove the function allIngredientsNames from the useEffect and get the array value in useMemo.
```
  const ingredientsNames = useMemo(()=>{
    let ingredientsNames: any = [];
    if (user.id) {
      for (let i = 1; i < ingredientArray.length; i++) {
        ingredientsNames.push(ingredientArray[i].name);
      }
      return ingredientsNames.join(",+");
    }
  },[user.id])
```

File: src/components/RecipesFromIngredients/RecipesFromIngredients.tsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+    if (user.id) {
+      for (let i = 1; i < ingredientArray.length; i++) {
+        ingredientsNames.push(ingredientArray[i].name);
+      }
+      return ingredientsNames.join(",+");
+    }
+  }
+
+  // get the id for the recipe.
+  const id: any =
```
Comment (Bderiel) [inline src/components/RecipesFromIngredients/RecipesFromIngredients.tsx:39]: this looks like array. Should at least type like

```const id: any[] = <array>```

File: src/components/RecipesFromIngredients/RecipesFromIngredients.tsx [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+      // this data will include title and id and other data about the recipe and ingredients.
+      // take the id and search another endpoint to extract more instructions about the recipes.
+      setRecievedRecipes(data);
+      console.log(
+        `******* This is the path: https://api.spoonacular.com/recipes/${id}/information?includeNutrition=false&apiKey=9b79c69e5e8943b88f963408824ce4a4`
+      );
+      if (id) {
+        // pass the id of the recipe to another endpoint to get the recipe data.
+        const recipeInfo = await axios.get(
+          `https://api.spoonacular.com/recipes/${id}/information?includeNutrition=false&apiKey=9b79c69e5e8943b88f963408824ce4a4`
```
Comment (Bderiel) [inline src/components/RecipesFromIngredients/RecipesFromIngredients.tsx:60]: Should add helpers to deal with api calls. like in a new file have a function getRecipes and other used API calls. You could abstract the API key into a variable and reuse it, this way if you need to change it you don't have to replace it in multiple places

File: src/components/RecipesFromIngredients/RecipesFromIngredients.tsx [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+          <div className="recipe_box">
+            <h1 className="recipe_h1">Recipe based on What You Have</h1>
+            <h2 className="recipe_h2">{recievedRecipesInfo.title}</h2>
+            <img className="recipe_img" src={imageURL} />
+            <h2 className="recipe_h2">Ingredients</h2>
+            <div className="recipe_ingredients">
+              {recievedRecipesInfo.extendedIngredients.map(
+                (foodItem: any, index: number) => {
+                  return (
+                    <div>
```
Comment (Bderiel) [inline src/components/RecipesFromIngredients/RecipesFromIngredients.tsx:99]: missing key

File: src/components/RecipesFromIngredients/RecipesFromIngredients.tsx (hunk 3 above)
Comment (antongeny) [reply src/components/RecipesFromIngredients/RecipesFromIngredients.tsx:61]: We also need to change the useEffect so it's not infinitely looping a request. It's the reason why we used up our daily amount of requests. Code change does not show here though.

File: src/components/RecipesFromIngredients/RecipesFromIngredients.tsx (hunk 1 above)
Comment (antongeny) [reply src/components/RecipesFromIngredients/RecipesFromIngredients.tsx]: Thank you so much for the commentary!

**Review by antongeny — APPROVED** [review]

**Pull request merged**